mmpose/models/losses/mesh_loss.py

```diff
@@ -0,0 +1,339 @@
+import torch
+import torch.nn as nn
+
+from ..mesh_heads.geometric_layers import batch_rodrigues
+from ..registry import LOSSES
+
+
+def perspective_projection(points, rotation, translation, focal_length,
+                           camera_center):
+    """This function computes the perspective projection of a set of 3D points.
+
+    Notes:
+        batch size: B
+        point number: N
+
+    Args:
+        points (Tensor([B, N, 3])): A set of 3D points
+        rotation (Tensor([B, 3, 3])): Camera rotation matrix
+        translation (Tensor([B, 3])): Camera translation
+        focal_length (Tensor([B,])): Focal length
+        camera_center (Tensor([B, 2])): Camera center
+
+    Returns:
+        projected_points (Tensor([B, N, 3])): Projected 2D
+            points in image space.
+    """
+
+    batch_size = points.shape[0]
+    K = torch.zeros([batch_size, 3, 3], device=points.device)
+    K[:, 0, 0] = focal_length
+    K[:, 1, 1] = focal_length
+    K[:, 2, 2] = 1.
+    K[:, :-1, -1] = camera_center
+
+    # Transform points
+    points = torch.einsum('bij,bkj->bki', rotation, points)
+    points = points + translation.unsqueeze(1)
+
+    # Apply perspective distortion
+    projected_points = points / points[:, :, -1].unsqueeze(-1)
+
+    # Apply camera intrinsics
+    projected_points = torch.einsum('bij,bkj->bki', K, projected_points)
+    projected_points = projected_points[:, :, :-1]
```

Comment:
what size is the last dim of projected_points

Reply:
It's the z coordinates. In other words, it's the depth.

Reply:
i mean, in L24, it says the returned shape is `[B, N, 3]`. but i guess the `[:, :, :-1]` might return some shape of `B, N, 2`

Reply:
Sorry, it's a typo.  The returned shape should be [B, N, 2]. I will fix it.